### ISSUE TITLE
Sketch launching on second display that's not currently in use

### ISSUE THREAD

**Issue by josephmoore** [processing/processing#3082]
I'm on OSX 10.8.5 and I'm unable to launch a sketch (no window is created in Java mode). However, I am able to write a program that prints to the console (still no window created). This is happening for me with Processing 2.2.1 and Processing 3.0a5. I'm running the following Java version:
java version "1.7.0_76"
Java(TM) SE Runtime Environment (build 1.7.0_76-b13)
Java HotSpot(TM) 64-Bit Server VM (build 24.76-b04, mixed mode)
I don't see any errors logging in the console and I've tried the various fixes mentioned: deleting the folder  ~/Library/Processing, deleting the folder ~/Documents/Processing. This just recently happened (within the past few days or so). Could it have been after the Java 7 Update 71? Any other thoughts?

**Comment by codyclay88** [comment]
Hey I just ran into this problem too. Turns out I just wasnt making the sketch visible. On your dock you should see a that a java application is running, try right-clicking it, go to options, and click display on [your display]. I had this problem because I use a thunderbolt display plugged in to my laptop and I guess it didnt know which display to put it on.

**Comment by josephmoore** [comment]
Thanks @codyclay88! I was doing the exact same thing!

**Comment by benfry** [comment]
Was the sketch not defaulting to the screen that the editor was on?

**Comment by codyclay88** [comment]
No it wasnt for some reason. It was almost like the sketch was confused so just had a nervous breakdown and did nothing.

**Comment by josephmoore** [comment]
Likewise, the sketch window was never created. In my case I had my computer plugged into a second monitor but the monitor was off. With the monitor on, the window is created on the second monitor. Additionally, I'm using a laptop (macbook pro).

**Comment by benfry** [comment]
Ok, that sounds like a bug. I use a setup like that and haven't run into it (with 3.0a5 or the latest from here on Github), but handling multiple displays is really tricky. 

Had either of you ever set the preference for which screen to use to start the sketch?

**Comment by josephmoore** [comment]
I didn't set anything in preferences. This happened with a clean install of both Processing 2.2.1and 3.0a5 (from processing.org). I also reinstalled the JDK and JRE each time. Let me know if I can send any other info.

**Comment by benfry** [comment]
Got it, thanks for following up. (FWIW, installing a JRE or JDK should have zero effect—Processing uses its own embedded JRE.)

**Comment by pjryan93** [comment]
I'm running the latest version of processing from here on github.
I'm on a late 2011 macbook pro running 10.10.2
Using an external Apple Cinema HD Display

I think I'm zeroing in on this problem.  I noticed this bug isn't isolated to when the screen is off.
I've noticed two conditions in resulting in no or strange placement of the sketch. The error persists regardless of the placement of the sketch window last launched.  However everything works fine when Full-screen is enabled.  

1)  In  preferences,  my display is set to 1(external display)  and the editor is located on display 2(my macbook pro).  So the main in the sketch is setting the window location, but the sketch never appears.
  
2)My editor is located on 1(external display) and the sketch is set to be displayed on display 1(the external display).  It registers that the Editor and RunDevice are on the same display, but the sketch doesn't appear the majority of the time or it appears on the wrong display.  I noticed the editor's coordinates being passed into PApplet are negative.  When explicitly setting the location of the frame it still shows up on the wrong display.

Is the external display  suppose to be index 1 in preferences and the default display?
